File: 0822-unique-morse-code-words/0822-unique-morse-code-words.py
class Solution(object):
    def uniqueMorseRepresentations(self, words):
        morse_codes = [
            ".-","-...","-.-.","-..",".","..-.","--.","....","..",".---",
            "-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-",
            "..-","...-",".--","-..-","-.--","--.."
        ]
        
        unique_transformations = set()
        
        for word in words:
            full_transformation = "".join(morse_codes[ord(char) - ord('a')] for char in word)
            unique_transformations.add(full_transformation)
            
        return len(unique_transformations)
        

File: 0822-unique-morse-code-words/0822-unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        final String[] morse = {
            ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---",
            "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-",
            "..-", "...-", ".--", "-..-", "-.--", "--.."
        };

        Set<String> transformations = new HashSet<>();
        
        for (final String word : words) {
            StringBuilder transformation = new StringBuilder();
            for (final char c : word.toCharArray()) {
                int index = c - 'a';
                transformation.append(morse[index]);
            }
            transformations.add(transformation.toString());
        }

        return transformations.size();
    }
}